### ISSUE TITLE
http 449 eror on sync

### ISSUE THREAD

**Issue by j-5-s** [dexie/Dexie.js#1803]
Looking at the docs [here](https://dexie.org/cloud/docs/access-control#example-a-simple-project-management-model) on the project manager code I implemented something similar.  However, i was getting a HTTP 449 on sync after executing something like this

```tsx
const realmId = await db.realms.add({
  name: `Team ${name}`,
  represents: "A baseball team",
});

const id = await db.teams.add({
  name,
  realmId,
  organization,
  playerIds: [],
  gameIds: [],
});
```

I needed to update the code to the following for it to work:
```tsx
  const realmId = await db.realms.add({
    name: `Team ${name}`,
    represents: "A baseball team",
  });

  await db.members.add({
    realmId,
    roles: ["manager"],
  });

  const id = await db.teams.add({
    name,
    realmId,
    organization,
    playerIds: [],
    gameIds: [],
  });
```

Although that resolves the issue im a bit confused here. 

1. I'm not specifying an email or userId in db.members.add, just the realmId that was just created with the "manager" role, is it using the the userId that is currently logged in by default? I assume when trying to share I need to add the other userId (which is the email i think). My roles and schema are below.
2. If a user creates a team object without logging in, then logs in, i was expecting the team object to persist. However the previously created team data is gone (as well as members) after logging and sync occurs and the unauthorized user is not merged with the logged in user.  Is it possible to do that?

```json
{
  "roles": {
    "manager": {
      "displayName": "Manager",
      "description": "Members with this role gains full permissions within the realm pointed out by the member entry",
      "sortOrder": 1,
      "permissions": { "manage": "*" }
    },
    "coach": {
      "displayName": "Can edit data but cant manage",
      "description": "Members with this role can create friends within a realm and update whether a friend is a good or not",
      "sortOrder": 2,
      "permissions": {
        "add": "*",
        "update": "*"
      }
    },
    "readonly": {
      "displayName": "ReadOnly",
      "description": "Members with this role have no permissions to change any data",
      "sortOrder": 3,
      "permissions": {}
    }
  }
}

```

```tsx
export class GamePlanner extends Dexie {
  teams!: DexieCloudTable<ITeam, "id">;
  players!: DexieCloudTable<IPlayer, "id">;
  games!: DexieCloudTable<IGame, "id">;
  innings!: DexieCloudTable<IInning, "id">;

  constructor() {
    super("GamePlanner", {
      addons: [dexieCloud],
      cache: "immutable",
    });
    this.version(1).stores({
      teams: "@id, name, organization, *playerIds, *gameIds",
      players: "@id, firstName, lastName, number, *teamIds",
      games: "@id, date, *inningIds, *playingIds, *notPlayingIds, updatedAt",
      innings: "@id, *dugoutPlayerIds, *fieldPlayerIds, gameId",

      // access control
      realms: "@realmId",
      members: "@id",
      roles: "[realmId+name]",
    });

    this.cloud.configure({
      databaseUrl: process.env.NEXT_PUBLIC_DEXIE_CLOUD_URL!,
      tryUseServiceWorker: true, // true!
      requireAuth: false,
    });
  }
}
```

**Comment by dfahlander** [comment]
Thanks! Looking in to this...

**Comment by dfahlander** [comment]
Issue confirmed. It happens when creating a realm without members. Issue solved in staging (version 1.2.8).
I've lifted your (@j-5-s) databases to staging so you should be able to verify it now. Let me know if you create a new db and need it to have the fix.
I'll wait some days before lifting production to latest version and switching staging db's back to prod, but probably within a few days. If there's someone else in need for this fix asap, please let me know.

**Comment by j-5-s** [comment]
thanks @dfahlander. No rush on my end. So that i understand whats happening though, the 449 error shouldn't happen when just creating a new realm and assigning it to another object?

**Comment by dfahlander** [comment]
Exactly, that's totally correct. In a previous version the flow required was to also explicitly add a member that connected your own userId to the new realm. This requirement was taken away and instead managed automatically on the server - it creates the "self member" automatically for the user that created the realm so there's no need to do so, but the issue was a consistency check later in the flow that compared the original request with the actual mutations and detected this inconsistency and threw the 449 error. So it's perfectly ok to create new realms without creating members.

**Comment by j-5-s** [comment]
thanks! I think I understand: Is it correct to think about the user who creates a realm as a the "owner" member which gives manager level privileges and not necessary to add any permissions or roles to that member.  That said, 

> If a member is removed from a realm, objects connected to that realm that are owned by the user will go out of scope for the use

I seem to have 5 or 6 realms that i have removed my user from and no longer have access to those scopes.  Is there a way on the cli or other interface to cleanup data?

**Comment by dfahlander** [comment]
Correct. You have full access on the entire DB only when using the CLI or the the REST API's [POST /all/members](https://dexie.org/cloud/docs/rest-api#post). Use [dexie-cloud export](https://dexie.org/cloud/docs/cli#export) to export an individual realm. Example:

```
dexie-cloud export --data --realmId "rlm~tdl0Ofo426bilS6y3fWMn6BNqA|zed" exportfile.json
```

Use a text editor to change the properties of one of the existing members or add a new one (ID must start with 'rlm' but could be of any length and chars). Use [dexie-cloud import](https://dexie.org/cloud/docs/cli#import) to import it back.

```
dexie-cloud import exportfile.json
```

If you prefer using the REST api, you can create new objects and let the server can generate their primary keys (on '@' based tables):

### Create new member
```
POST /all/members HTTP/1.1
Authorization: Bearer xxxxxx
Content-Type: application/json

[{
  "realmId": <theRealmId>,
  "userId": <yourUserId>,
  "roles": ["manager"]
}]
```

### Update existing
```
POST /all/members HTTP/1.1
Authorization: Bearer xxxxxx
Content-Type: application/json

[{
  "id": <memberId>,
  "realmId": <theRealmId>,
  "userId": <yourUserId>,
  "roles": ["manager"]
}]
```


*FYI: Production is now on version 1.2.8 also so the fix has reached all customers and your dbs is back in the production slot.*
